sonic-thermalctld/scripts/thermalctld

```diff
@@ -119,57 +119,35 @@ class FanStatus(logger.Logger):
         self.status = status
         return True
 
-    def _check_speed_value_available(self, speed, target_speed, tolerance, current_status):
-        if speed == NOT_AVAILABLE or target_speed == NOT_AVAILABLE or tolerance == NOT_AVAILABLE:
-            if isinstance(tolerance, int) and (tolerance > 100 or tolerance < 0):
-                self.log_warning('Invalid tolerance value: {}'.format(tolerance))
-                return False
-
-            if current_status is True:
-                self.log_warning('Fan speed or target_speed or tolerance became unavailable, '
-                                 'speed={}, target_speed={}, tolerance={}'.format(speed, target_speed, tolerance))
-            return False
-        return True
-
-    def set_under_speed(self, speed, target_speed, tolerance):
+    def set_under_speed(self, is_under_speed):
         """
         Set and cache Fan under speed status
-        :param speed: Fan speed
-        :param target_speed: Fan target speed
-        :param tolerance: Threshold between Fan speed and target speed
+        :param is_under_speed: Fan under speed threshold status
         :return: True if status changed else False
         """
-        if not self._check_speed_value_available(speed, target_speed, tolerance, self.under_speed):
-            old_status = self.under_speed
-            self.under_speed = False
-            return old_status != self.under_speed
+        if is_under_speed == NOT_AVAILABLE:
+            if self.under_speed:
+                self.log_warning('Fan under speed threshold check became unavailable')
+            is_under_speed = False
```

Comment:
Under which scenario would this warning be seen? Is this check necessary?

Reply:
If a platform implementation of `is_under_speed()` raises an exception, it is caught and a warning is provided here. The condition of the log could perhaps be updated, I'll check that.

Reply:
yeah, that is true, it is an existing condition. Just wanted to understand the use case for the inner if condition